Database schema upgrade: Delete unused slug; test suite description column

okurz opened the pull request:
Needs database schema bump.

okurz commented:
can someone explain why I get the error message

```
t/deploy.t .. 1/? Can't call method "id" on an undefined value at t/deploy.t line 86 (#1)
```

now?

aaannz commented:
I think that the deploy test is not correct anymore. When upgrade is being tested, deploy test installs old schema version directly using deployment handler. This handler does not create system user as `_try_deploy_db` does. Then from that old version to your new one migration does not cover creation of system user, thus test fails.

coolo commented:
Then let's take https://github.com/os-autoinst/openQA/pull/1095 and have try_deploy_db take a version to deploy

coolo commented:
makes me wonder why the deploy.t isn't failing in #1094

coolo commented on lib/OpenQA/Schema/Result/JobTemplates.pm:
```diff
@@ -52,10 +52,10 @@ __PACKAGE__->belongs_to(test_suite => 'OpenQA::Schema::Result::TestSuites', 'tes
     my $self = shift;
 
-    $self->machine->variables . $self->test_suite->variables . $self->product->variables;
+    $self->machine->description . $self->test_suite->description . $self->product->description;
```
I don't think this simple concat makes much sense.

okurz commented:
Updated:
* rebased
* included #1095
* merged database schema

test fails now with `The command "openssl aes-256-cbc -K $encrypted_e2c381aa6b8c_key -iv $encrypted_e2c381aa6b8c_iv -in .openqa-travis.enc -out .openqa-travis -d" failed and exited with 1 during .` what is that?

coolo commented:
please rebase

okurz commented:
rebased, test still running, at least passed previous failure already
